### DISCUSSION TITLE
四五张千万或者上亿表做join效率太低了

### DISCUSSION THREAD

**Discussion by young138120** [apache/doris#26967] [A - General / Q&A]
现有几张大几千万以及部分上亿的表（部分有做分区，部分没有），一些几张维度表（这部分大概在几万左右）
大表做在一个 colocation join group里面（bucket只设置一个key），并且join字段都是unique key 并且是bucket key
然后做测试，发现效率很差，主表过滤完只剩一个分区，还需要十多秒
然后对所有的设计查询的表做了analyse
然后设置
set  enable_fallback_to_original_planner=false;
Set  forbid_unknown_col_stats=true;
Set  enable_nereids_planner=true;
set  enable_pipeline_engine=true;
还是需要十几秒，看了explain信息，都是用到了join op: INNER JOIN(COLOCATE[])

doris version：2.0.0

**Comment by young138120** [comment]
同样的语句再mysql只需要几百毫秒，这差距也太大了吧。
该做的优化我已经尽量做了，而且是走colocation join了

**Reply by young138120** [reply]
already confirmed it is a bug，and will be fix in 2.1.0 version   
thank you  for you reply

**Reply by young138120** [reply]
but i don't know the real query efficiency  in new version    
 can you confirm the efficiency  with mysql ?

**Reply by LemonLiTree** [reply]
I don't have your case
